### DISCUSSION TITLE
Why lsp is not showing all possible code actions for current line?

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2563] [Q&A] [answered]
Take a look:

https://github.com/LazyVim/LazyVim/assets/51330172/845aa3ae-5910-45cd-9144-b16d2bfedb01

Why code actions telescope is not showing all possible code actions for current line? Is it possible to get all LSP actions for the current line? 

It is also quite interesting how telescope LSP picks available actions if they differ from LSP's ones. Does somebody know?

**Comment by dpetka2001** [comment, marked as answer]
I don't know much about LSP specification. But from quickly browsing the Microsoft LSP specification it seems that `rename` does not fall under code actions. When you want to rename you send a request [textDocument.rename](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_rename) whereas for code actions you send a request [textDocument/codeAction](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeAction). So, it seems that the 2 are different entities under the LSP specification.

**Reply by Demianeen** [reply]
Oh, my, it is working with code rename action on `<leader>cr` in Lazyvim! Thanks)
